Content.Client/Graphics/Overlays/GradientCircleMask.cs

```diff
@@ -13,18 +15,43 @@ public class GradientCircleMaskOverlay : Overlay
     {
         [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
         [Dependency] private readonly IEyeManager _eyeManager = default!;
+        [Dependency] private readonly IPlayerManager _playerManager = default!;
 
         public override OverlaySpace Space => OverlaySpace.WorldSpace;
         private readonly ShaderInstance _shader;
 
-        public GradientCircleMaskOverlay() : base(nameof(SharedOverlayID.GradientCircleMaskOverlay))
+        public GradientCircleMaskOverlay() : base(nameof(GradientCircleMaskOverlay))
         {
             IoCManager.InjectDependencies(this);
             _shader = _prototypeManager.Index<ShaderPrototype>("GradientCircleMask").Instance();
         }
 
+        public static bool LocalPlayerHasState(IPlayerManager pm, bool critical, bool dead) {
+            var playerEntity = pm.LocalPlayer?.ControlledEntity;
+
+            if (playerEntity == null)
+            {
+                return false;
+            }
+
+            if (playerEntity.TryGetComponent<IMobStateComponent>(out var mobState))
+            {
+                if (critical)
+                    if (mobState.IsCritical())
+                        return true;
+                if (dead)
+                    if (mobState.IsDead())
+                        return true;
+            }
+
+            return false;
+        }
```

Comment:
Shouldn't you just be checking the mobstates? Ideally whatever manages the crit / death masks just listens for mobstate events and then adjusts accordingly.

Reply:
See the same basic thing I put on all the other cases like this